x-pack/plugins/security_solution/public/common/components/sourcerer/temporary.tsx

```diff
@@ -185,4 +186,58 @@ export const TemporarySourcerer = React.memo<Props>(
   }
 );
 
+TemporarySourcererComp.displayName = 'TemporarySourcererComp';
+
+interface TemporarySourcererProps {
+  activePatterns?: string[];
+  indicesExist: boolean;
+  isModified: 'deprecated' | 'missingPatterns';
+  isShowingUpdateModal: boolean;
+  missingPatterns: string[];
+  onContinueWithoutUpdate: () => void;
+  onDismiss: () => void;
+  onDismissModal: () => void;
+  onReset: () => void;
+  onUpdateStepOne: () => void;
+  onUpdateStepTwo: () => void;
+  selectedPatterns: string[];
+}
+
+export const TemporarySourcerer = React.memo<TemporarySourcererProps>(
+  ({
+    activePatterns,
+    indicesExist,
+    isModified,
+    missingPatterns,
+    onContinueWithoutUpdate,
+    onDismiss,
+    onReset,
+    onUpdateStepOne,
+    onUpdateStepTwo,
+    selectedPatterns,
+    isShowingUpdateModal,
+    onDismissModal,
+  }) => (
+    <>
+      <TemporarySourcererComp
```

Comment:
changes in this file include renaming the functions to be more descriptive, and moving this combined component from`components/sourcerer/index.tsx`